jlm/mlir/frontend/MlirToJlmConverter.cpp

```diff
@@ -268,20 +314,52 @@ MlirToJlmConverter::ConvertOperation(
 {
 
   // ** region Arithmetic Integer Operation **
-  auto convertedNode = ConvertBitBinaryNode(mlirOperation, inputs);
+  auto convertedBitBinaryNode = ConvertBitBinaryNode(mlirOperation, inputs);
   // If the operation was converted it means it has been casted to a bit binary operation
-  if (convertedNode)
-    return convertedNode;
+  if (convertedBitBinaryNode)
+    return convertedBitBinaryNode;
   // ** endregion Arithmetic Integer Operation **
 
+  // ** region Arithmetic Float Operation **
+  auto convertedFloatBinaryNode = ConvertFPBinaryNode(mlirOperation, rvsdgRegion, inputs);
+  // If the operation was converted it means it has been casted to a fp binary operation
+  if (convertedFloatBinaryNode)
+    return convertedFloatBinaryNode;
+  // ** endregion Arithmetic Float Operation **
+
   if (auto castedOp = ::mlir::dyn_cast<::mlir::arith::ExtUIOp>(&mlirOperation))
   {
     auto st = dynamic_cast<const jlm::rvsdg::bittype *>(&inputs[0]->type());
     if (!st)
-      JLM_ASSERT("frontend : expected bitstring type for ExtUIOp operation.");
+      JLM_UNREACHABLE("Expected bitstring type for ExtUIOp operation.");
     ::mlir::Type type = castedOp.getType();
     return rvsdg::output::GetNode(*&llvm::zext_op::Create(*(inputs[0]), ConvertType(type)));
   }
+  else if (auto castedOp = ::mlir::dyn_cast<::mlir::arith::ExtSIOp>(&mlirOperation))
+  {
+    auto outputType = castedOp.getOut().getType();
+    auto convertedOutputType = ConvertType(outputType);
+    if (!::mlir::isa<::mlir::IntegerType>(castedOp.getType()))
+      JLM_UNREACHABLE("Expected IntegerType for ExtSIOp operation output.");
+    return rvsdg::output::GetNode(*llvm::sext_op::create(
+        static_cast<size_t>(castedOp.getType().cast<::mlir::IntegerType>().getWidth()),
+        inputs[0]));
+  }
+  else if (auto sitofpOp = ::mlir::dyn_cast<::mlir::arith::SIToFPOp>(&mlirOperation))
+  {
+    auto st = std::dynamic_pointer_cast<const jlm::rvsdg::bittype>(inputs[0]->Type());
```

Comment:
Since st doesn't seem to be used. Isn't it possible to simply put the dynamic cast in the if statement?

Reply:
st is used on line 357